src/lib/storage/chunk.cpp

```diff
@@ -15,26 +15,26 @@
 namespace opossum {
 
 void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) {
-  // Implementation goes here
+  Assert(size() == 0 || segment->size() == size(), "Segment needs to have same number of rows as chunk");
+  _segments.push_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "New values need same number of values as chunk columns");
+  for (auto i = 0ul, end = values.size(); i < end; i++) {
```

Comment:
`values.size()` is evaluated in each iteration - save `values.size()` to variable and use it instead